File: app/src/main/java/com/example/curity/SignUp/SignUpP1.java
package com.example.curity.SignUp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.curity.R;
import com.google.firebase.auth.FirebaseAuth;

public class SignUpP1 extends AppCompatActivity {

    private EditText firstEditText, lastEditText, addressEditText, phoneEditText;
    private Button nxtBtn;
    private FirebaseAuth firebaseAuth;
    private Boolean check = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up_p1);

        firstEditText = findViewById(R.id.f_nameEt);
        lastEditText = findViewById(R.id.l_nameEt);
        addressEditText = findViewById(R.id.addressEt);
        phoneEditText = findViewById(R.id.phoneEt);

        nxtBtn = findViewById(R.id.button1);

        nxtBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final String firstTxt = firstEditText.getText().toString();
                final String lastTxt = lastEditText.getText().toString();
                final String addressTxt = addressEditText.getText().toString();
                final String phoneTxt = phoneEditText.getText().toString();

                if (firstTxt.isEmpty()){
                    firstEditText.setError("First name is required");
                    firstEditText.requestFocus();
                }

                if (lastTxt.isEmpty()){
                    lastEditText.setError("last name is required");
                    lastEditText.requestFocus();
                }

                if (addressTxt.isEmpty()){
                    addressEditText.setError("Address is required");
                    addressEditText.requestFocus();
                }

                if (phoneTxt.isEmpty()){
                    phoneEditText.setError("Cellphone number is required");
                    phoneEditText.requestFocus();
                }

                if (phoneTxt.length() != 11){
                    phoneEditText.setError("Enter the 11 digits of your cellphone number.");
                    phoneEditText.requestFocus();
                }

                if (!(firstTxt.isEmpty() || lastTxt.isEmpty() || addressTxt.isEmpty() || phoneTxt.isEmpty()) && phoneTxt.length() == 11) {
                    //pass the data to the next page
                    Intent intent = new Intent(getApplicationContext(), SignUpP2.class);
                    intent.putExtra("f_name", firstTxt);
                    intent.putExtra("l_name", lastTxt);
                    intent.putExtra("address", addressTxt);
                    intent.putExtra("phone", phoneTxt);

                    startActivity(intent);
                    finish();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/curity/SignUp/SignUpP1.kt
package com.example.curity.SignUp

import android.content.ContentValues.TAG
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.curity.Login
import com.example.curity.databinding.ActivitySignUpP1Binding
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.UserProfileChangeRequest


class SignUpP1 : AppCompatActivity() {

    private lateinit var binding: ActivitySignUpP1Binding
    private lateinit var firebaseAuth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySignUpP1Binding.inflate(layoutInflater)
        setContentView(binding.root)


        firebaseAuth = FirebaseAuth.getInstance()

        binding.textView1.setOnClickListener{
            val intent = Intent(this, Login::class.java)
            startActivity(intent)
        }

        binding.button1.setOnClickListener {
            val email = binding.emailEt.text.toString()
            val pass = binding.passET.text.toString()
            val confirmPass = binding.confirmPassEt.text.toString()
            val fName = binding.fNameEt.text.toString()
            val lName = binding.lNameEt.text.toString()


            if (email.isNotEmpty() && pass.isNotEmpty() && confirmPass.isNotEmpty()
                && fName.isNotEmpty() && lName.isNotEmpty()) {
                if (pass == confirmPass){
                    firebaseAuth.createUserWithEmailAndPassword(email,pass).addOnCompleteListener{
                        if (it.isSuccessful){
                            val intent = Intent(this, SignUpP2::class.java)

                            val user = FirebaseAuth.getInstance().currentUser
                            val fullName = fName.plus(" ").plus(lName)

                            val profileUpdates = UserProfileChangeRequest.Builder()
                                .setDisplayName(fullName).build()

                            user!!.updateProfile(profileUpdates)
                                .addOnCompleteListener { task ->
                                    if (task.isSuccessful) {
                                        Log.d(TAG, "User profile updated.")
                                    }
                                }
                            startActivity(intent)
                        } else {
                            Toast.makeText(this, it.exception.toString(), Toast.LENGTH_SHORT).show()
                        }
                    }

                } else {
                    Toast.makeText(this, "Password is not matched", Toast.LENGTH_SHORT).show()
                }
            }else{
                Toast.makeText(this, "Please fill Up the empty field(s)", Toast.LENGTH_SHORT).show()
            }
        }
    }
}